examples/iterators.cpp

```diff
@@ -13,19 +12,19 @@ int main()
         DynamicData str1(string);
 
         str1.value<std::string>("Hello!");
-        for (ReadableDynamicDataRef&& elem : str1)
+        for (auto&& elem : str1)
```

Comment:
This is an example, using `auto` obfuscates the code making it more difficult to understand.